### DISCUSSION TITLE
Expected behavior of Multi-select with `extendSelection`

### DISCUSSION THREAD

**Discussion by jaknas** [adobe/react-spectrum#4947] [Q&A] [answered]
Hi, I wanted to ask about expected behavior of the [extendSelection](https://github.com/adobe/react-spectrum/blame/8987f9283871b1bda0acc4637096966275f82167/packages/%40react-stately/selection/src/SelectionManager.ts#L207) method.

As a live example I will use [this code sandbox](https://codesandbox.io/s/lucid-fire-mf0m6j?file=/src/App.js).

Reproduction steps:

1. Select "Games" with Space
2. Unselect "Games" with Space
3. Press down arrow 2 times, to move to "bootmgr"
4. Hold Shift and press down arrow.

**Expected:** "bootmgr" and "log.txt" will be selected
**Actual:** All rows are selected.

More precisely, I think the question is whether the `anchorKey` should work the same when there are no selected rows. For me, it is a bit confusing to see all rows being selected.

Wanted to know your thoughts about this and to know if there's a particular reason it works like this.

Thanks!

**Comment by snowystinger** [comment, marked as answer]
I agree, some work should be done here. I think the select + unselect then shift is a bug, if you don't mind opening an issue. This can be observed in our docs as well https://react-spectrum.adobe.com/react-spectrum/ListView.html#example

I think the `anchorKey` should be set to the `focusedKey` when the shift key is pressed. I'm not sure about non-contiguous multi select, if shift should *always* add to the current selection. But we can at least address the `anchorKey`.

We accept PR's if you feel like making one.

**Reply by jaknas** [reply]
Thanks. I'm not sure If I fully understand:

 > I'm not sure about non-contiguous multi select, if shift should always add to the current selection.

But we can continue in the issue I created: #4952

**Reply by snowystinger** [reply]
I'll keep this a separate discussion for now. The general gist is, let's say you have items 1-6.
1) select item 1
2) keyboard navigate to item 4
3) press and hold shift key
4) keyboard navigate to item 5

is the expectation that the items selected would be 1, 4, 5
or is the expectation that the items selected would be 1, 2, 3, 4, 5

I feel like it should be the first option, and I think the recommendation I gave to set the anchorKey when shift is pressed would yield this result. But it's somewhat different to how we currently work, which is closer to the second option

**Reply by jaknas** [reply]
> I'll keep this a separate discussion for now. The general gist is, let's say you have items 1-6.
> 
> 1) select item 1
> 
> 2) keyboard navigate to item 4
> 
> 3) press and hold shift key
> 
> 4) keyboard navigate to item 5
> 
> 
> 
> is the expectation that the items selected would be 1, 4, 5
> 
> or is the expectation that the items selected would be 1, 2, 3, 4, 5
> 
> 
> 
> I feel like it should be the first option, and I think the recommendation I gave to set the anchorKey when shift is pressed would yield this result. But it's somewhat different to how we currently work, which is closer to the second option

Oh I get it now. I agree that first option makes the most sense.  TBH I got accustomed to the second option and I learned to do a workaround to reset the anchorKey, if I want to do non contiguous selection.

I feel like the first one seems the easiest to understand w.r.t how it works
